zrml/prediction-markets/src/lib.rs

```diff
@@ -1376,59 +1458,51 @@ mod pallet {
             Ok(())
         }
 
-        // Check that the market hasn't reached the end of its period yet.
         fn ensure_market_is_active(
-            period: &MarketPeriod<T::BlockNumber, MomentOf<T>>,
+            market: &Market<T::AccountId, T::BlockNumber, MomentOf<T>>,
         ) -> DispatchResult {
-            ensure!(
-                match period {
-                    MarketPeriod::Block(range) => {
-                        <frame_system::Pallet<T>>::block_number() < range.end
-                    }
-                    MarketPeriod::Timestamp(range) => {
-                        T::MarketCommons::now() < range.end
-                    }
-                },
-                Error::<T>::MarketIsNotActive
-            );
+            ensure!(market.status == MarketStatus::Active, Error::<T>::MarketIsNotActive);
             Ok(())
         }
```

Comment:
Yeah, I agree. However, in the near future this will probably become a method of `struct Market`. So I'd rather keep it in the abstract form it is now. Should make it easier to switch to using the class method.